pkg/utils/resourceid/owncloud.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package resourceid
+
+import (
+	"encoding/base64"
+	"errors"
+	"strings"
+	"unicode/utf8"
+
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+)
+
+const (
+	idDelimiter string = ":"
+)
+
+// OwnCloudResourceIDUnwrap returns the wrapped resource id
+// by OwnCloudResourceIDWrap and returns nil if not possible
+func OwnCloudResourceIDUnwrap(rid string) *provider.ResourceId {
+	parts, err := unwrap(rid)
+	if err != nil {
+		return nil
+	}
+
+	return &provider.ResourceId{
+		StorageId: parts[0],
+		OpaqueId:  parts[1],
+	}
+}
+
+func unwrap(rid string) ([]string, error) {
```

Comment:
Let's return the resource ID here itself?

Reply:
Done :+1: